rust/chains/hyperlane-ethereum/src/interchain_gas.rs

```diff
@@ -130,6 +125,17 @@ where
     }
 }
 
+#[async_trait]
+impl<M> NonceIndexer<InterchainGasPayment> for EthereumInterchainGasPaymasterIndexer<M>
+where
+    M: Middleware + 'static,
+{
+    async fn nonce_at_tip(&self) -> ChainResult<(u32, u32)> {
+        info!("Gas payment nonce indexing not implemented");
```

Comment:
panicking here because it should never be used in Ethereum (at least not rn) seems reasonable to me